__tests__/app.test.js

```diff
@@ -138,4 +137,49 @@ describe('GET: /api/articles/:article_id/comments', () => {
             expect(msg).toBe("Bad request")
         })
     })
+});
+describe('POST: /api/articles/:article_id/comments', () => {
+    it('201: responds with correct status and adds a comment object to article with correct properties (username and body)', () => {
+
+        const newComment = {username: 'butter_bridge', body: 'this is a comment'}
+        return request(app).post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(201)
+        .then(({body}) => {
+            const {comment} = body;
+            expect(comment).toMatchObject({comment_id: 19, article_id:1, votes:0, author: 'butter_bridge', body: 'this is a comment'})
+        })
+    });
+    it('404: sends an appropriate error message when username is invalid', () => {
+        const newComment = {username: 'buttery_bridge', body: 'this is a comment'}
+        return request(app).post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(404)
+        .then(({body}) => {
+            const {msg} = body;
+            expect(msg).toBe('Not found')
+
+        })
+    });
+    it('404: sends an appropriate error message when given a valid but non existent article_id', () => {
+        const newComment = {username: 'butter_bridge', body: 'this is a comment'}
+        return request(app).post('/api/articles/999/comments')
+        .send(newComment)
+        .expect(404)
+        .then(({body}) => {
+            const {msg} = body;
+            expect(msg).toBe('Not found')
+        })
+    })
+    it('400: sends an appropriate error message when given an invalid article_id', () => {
+        const newComment = {username: 'butter_bridge', body: 'this is a comment'}
+        return request(app).post('/api/articles/dog/comments')
+        .send(newComment)
+        .expect(400)
+        .then(({body}) => {
+            const {msg} = body;
+            expect(msg).toBe('Bad request')
+        })
```

Comment:
We should add a test for when we do not receive a username or a body, and send a 400 bad request 🙂